drivers/net/ethernet/intel/idpf/idpf_eth_idc.h

```diff
@@ -4,125 +4,153 @@
 #ifndef _IDPF_ETH_IDC_H_
 #define _IDPF_ETH_IDC_H_
 
-/**
- * enum idpf_vport_reset_cause - Vport soft reset causes
- * @IDPF_SR_Q_CHANGE: Soft reset queue change
- * @IDPF_SR_Q_DESC_CHANGE: Soft reset descriptor change
- * @IDPF_SR_MTU_CHANGE: Soft reset MTU change
- * @IDPF_SR_RSC_CHANGE: Soft reset RSC change
- */
-enum idpf_vport_reset_cause {
-	IDPF_SR_Q_CHANGE,
-	IDPF_SR_Q_DESC_CHANGE,
-	IDPF_SR_MTU_CHANGE,
-	IDPF_SR_RSC_CHANGE,
+#include <linux/auxiliary_bus.h>
+
+struct idpf_adapter;
```

Comment:
With idpf_dev_info_to_adapter macro gone you would not need the above line?

Reply:
It's forward declaration of idpf_adapter structure. It is needed in this file. We cannot include idpf.h, in that case this is the only way to make it happen. I am not sure what are you referring to?

Reply:
Done!